core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/interceptor/SyncHttpChecksumInTrailerInterceptor.java

```diff
@@ -90,14 +90,15 @@ public SdkHttpRequest modifyHttpRequest(Context.ModifyHttpRequest context, Execu
 
         ChecksumSpecs checksumSpecs =
             HttpChecksumUtils.checksumSpecWithRequestAlgorithm(executionAttributes).orElse(null);
-        if (checksumSpecs == null ||
-            !HttpChecksumUtils
-                .isTrailerBasedChecksumForClientType(
-                    executionAttributes,
-                    context.httpRequest(),
-                    ClientType.SYNC, checksumSpecs,
-                    context.requestBody().isPresent(),
-                    context.requestBody().map(requestBody -> requestBody.contentStreamProvider() != null).orElse(false))) {
+        if (checksumSpecs == null
+            || !context.requestBody().isPresent()
+            || !HttpChecksumUtils
+            .isTrailerBasedChecksumForClientType(
+                executionAttributes,
+                context.httpRequest(),
+                ClientType.SYNC, checksumSpecs,
+                context.requestBody().isPresent(),
+                context.requestBody().map(requestBody -> requestBody.contentStreamProvider() != null).orElse(false))) {
```

Comment:
This is a bit hard to read. Can we create a method for it?

Reply:
done